File: peak_element.java
// Time Complexity : O(log n)
// Space Complexity : 1
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :
// Your code here along with comments explaining your approach in three sentences only
class Solution {
    public int findPeakElement(int[] nums) {
        if(nums == null || nums.length == 0){
            return -1;
        }
        int low = 0;
        int high = nums.length-1;

        while(low <= high){
            int mid = low + (high-low)/2;
            if(mid > low && nums[mid] < nums[mid - 1]){
                    high = mid - 1;
            }
            else if(mid < nums.length-1  && nums[mid] < nums[mid + 1]){
                    low = mid + 1;
            }
            else{
                return mid;
            }
        }
        return -1;
    }
}

File: Peakelement.py
# Time Complexity : 0(logn)
# Space Complexity : 0(1)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : NA


class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        low = 0
        high = len(nums) -1 
        while low <= high:
            mid = low + (high - low )//2

            if (mid == 0 or nums[mid] > nums[mid - 1]) and (mid == high or nums[mid] > nums[mid + 1]):
                return mid

            elif nums[mid + 1] > nums[mid]:
                low = mid +1
            else:
                high = mid - 1